### ISSUE TITLE
GPU Driver Installation fails on latest Rocky 8 Image

### ISSUE THREAD

**Issue by wenyhu-google** [GoogleCloudPlatform/compute-gpu-installation#31]
Create an GCP Instance with latest Rocky linux 8 image: rocky-linux-8-v20231115.

Install GPU Driver following https://cloud.google.com/compute/docs/gpus/install-drivers-gpu#installation_scripts:

```
curl https://raw.githubusercontent.com/GoogleCloudPlatform/compute-gpu-installation/main/linux/install_gpu_driver.py --output install_gpu_driver.py
sudo python3 install_gpu_driver.py
```

GPU driver installation failed with error:

```
[2023-12-05 20:54:31] Executing: dnf install -y kernel-devel-4.18.0-477.27.1.el8_8.x86_64 kernel-headers-4.18.0-477.27.1.el8_8.x86_64

Error while executing `dnf install -y kernel-devel-4.18.0-477.27.1.el8_8.x86_64 kernel-headers-4.18.0-477.27.1.el8_8.x86_64`:
Command '['dnf', 'install', '-y', 'kernel-devel-4.18.0-477.27.1.el8_8.x86_64', 'kernel-headers-4.18.0-477.27.1.el8_8.x86_64']' returned non-zero exit status 1.
Failed with exception: Command '['dnf', 'install', '-y', 'kernel-devel-4.18.0-477.27.1.el8_8.x86_64', 'kernel-headers-4.18.0-477.27.1.el8_8.x86_64']' returned non-zero exit status 1.
Traceback (most recent call last):
  File "install_gpu_driver.py", line 461, in <module>
    raise err
  File "install_gpu_driver.py", line 458, in <module>
    main()
  File "install_gpu_driver.py", line 453, in main
    install(args)
  File "install_gpu_driver.py", line 431, in install
    install_dependencies(system, version)
  File "install_gpu_driver.py", line 343, in install_dependencies
    reboot_flag = install_dependencies_centos_rhel_rocky(system, version)
  File "install_gpu_driver.py", line 272, in install_dependencies_centos_rhel_rocky
    run(f"dnf install -y kernel-devel-{kernel_version} kernel-headers-{kernel_version}")
  File "install_gpu_driver.py", line 140, in run
    raise err
  File "install_gpu_driver.py", line 135, in run
    input=input, cwd=cwd, env=environment)
  File "/usr/lib64/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['dnf', 'install', '-y', 'kernel-devel-4.18.0-477.27.1.el8_8.x86_64', 'kernel-headers-4.18.0-477.27.1.el8_8.x86_64']' returned non-zero exit status 1.
```

The installation failed on `kernel_version=$(uname -r) && dnf install -y "kernel-devel-$kernel_version" "kernel-headers-$kernel_version"`, where the kernel_version is `4.18.0-477.27.1.el8_8.x86_64`.

**Comment by m-strzelczyk** [comment]
I tried running this system image and it seems that:

1. The system has kernel version `4.18.0-477.27.1.el8_8.x86_64`
2. `dnf search kernel-devel` gives only `kernel-devel.x86_64 : Development package for building kernel modules to match the kernel`
3. When I tried to install this `kernel-devel.x86_64` package it turns out to be `kernel-devel-4.18.0-513.9.1.el8_9.x86_64` under the hood.

As we can see, the versions don't match. I don't know if there's anything I can do to help with this issue. We will probably soon get a new version of the Rocky 8 image or an update in the repository and the versions will match again...
